File: sdks/csharp/src/CanaryGate/Hash.cs
namespace CanaryGate;

public static class Hash
{
    public static int HashString(string input)
    {
        var hash = 5381;
        for (var i = 0; i < input.Length; i++)
        {
            unchecked
            {
                hash = ((hash << 5) + hash) ^ input[i];
            }
        }
        return (int)((uint)hash % 100u);
    }
}

File: sdks/java/src/main/java/com/canarygate/Hash.java
package com.canarygate;

public final class Hash {

    private Hash() {
    }

    public static int hashString(String input) {
        int hash = 5381;
        for (int i = 0; i < input.length(); i++) {
            hash = ((hash << 5) + hash) ^ input.charAt(i);
        }
        return Integer.remainderUnsigned(hash, 100);
    }
}
